logstreams/src/main/java/io/zeebe/logstreams/state/StateSnapshotController.java

```diff
@@ -38,19 +34,32 @@
   private static final Logger LOG = Loggers.SNAPSHOT_LOGGER;
 
   private final StateStorage storage;
-  private final SnapshotReplication replication;
   private final ZeebeDbFactory zeebeDbFactory;
   private ZeebeDb db;
+  private final ReplicationController replicationController;
 
   public StateSnapshotController(final ZeebeDbFactory rocksDbFactory, final StateStorage storage) {
-    this(rocksDbFactory, storage, new NoneSnapshotReplication());
+    this(rocksDbFactory, storage, new NoneSnapshotReplication(), 1);
   }
 
   public StateSnapshotController(
-      ZeebeDbFactory zeebeDbFactory, StateStorage storage, SnapshotReplication replication) {
+      ZeebeDbFactory zeebeDbFactory,
+      StateStorage storage,
+      SnapshotReplication replication,
+      int maxSnapshots) {
     this.storage = storage;
-    this.replication = replication;
     this.zeebeDbFactory = zeebeDbFactory;
+    replicationController =
+        new ReplicationController(
+            replication,
+            storage,
```

Comment:
I get confused about all the different names: `StateSnapshotControler`, `SnapshotReplication`, `ReplicationController`, `AsyncSnapshotDirector`, etc. :sweat_smile: